File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        // (freq, val)
        priority_queue<pair<int, int>> heap;

        unordered_map<int, int> freq;

        for (auto i : nums) {
            freq[i]++;
        }

        for (auto& i : freq) {
            heap.push({i.second*-1, i.first*-1});
            if (heap.size() > k) {
                heap.pop();
            }
        }

        vector<int> res;

        for (int i = 0; i < k; i++) {
            res.push_back(heap.top().second * -1);
            heap.pop();
        }
        return res;
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-5.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        res = []
        l = 0
        r = 1
        length = len(temperatures)
        while r < length:
            if temperatures[r] > temperatures[l]:
                res.append(r-l)
                while stack and temperatures[r] > stack[-1][0]:
                    res[stack[-1][1]] = r - stack[-1][1]
                    stack.pop()
            else:
                res.append(0)
                stack.append([temperatures[l], l])
            r += 1
            l += 1
        res.append(0)
        return res